src/java/main/org/apache/zookeeper/server/NIOServerCnxn.java

```diff
@@ -1001,25 +1010,14 @@ public String toString() {
     @Override
     public void close() {
         synchronized(factory.cnxns){
```

Comment:
The removeCnxn already synchronizes on the cnxns so this synchronization can be removed.